activerecord/test/cases/autosave_association_test.rb

```diff
@@ -1774,3 +1775,29 @@ def test_after_save_callback_with_autosave
     assert_equal 1, comment.post_comments_count
   end
 end
+
+
+class TestAutosaveAssociationOnAHasManyAssociationDefinedINSubclass < ActiveRecord::TestCase
+  class Agency < Firm
+    has_many :projects, foreign_key: :firm_id
+
+    accepts_nested_attributes_for :projects
+  end
+
+  def test_should_update_children_when_asssociation_redefined_in_subclass
+    agency = Agency.create!(name: 'Agency')
+    valid_project = Project.create!(firm: agency, name: 'Initial')
+    agency.update!({
+      'projects_attributes' => {
+        '0' => {
+          'name' => 'Updated',
+          'id' => valid_project.id
+        }
+      }
+    })
+
+    valid_project.reload
+
+    assert_equal valid_project.name, 'Updated'
```

Comment:
Could you change order of arguments since `assert_equal` receive expected value as first argument and actual values as the second.
It would improve error message:
```
Expected: "Initial"
  Actual: "Updated"
```